daemon/list.go

```diff
@@ -61,6 +64,21 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			}
 		}
 	}
+
+	if ancestors, ok := psFilters["ancestor"]; ok {
+		ancestorFilter = true
+		byParents := daemon.Graph().ByParent()
+		// The idea is to walk the graph down the most "efficient" way.
+		for _, ancestor := range ancestors {
+			// First, get the imageId of the ancestor filter (yay)
+			image, err := daemon.Repositories().LookupImage(ancestor)
+			if err == nil {
```

Comment:
Oh right, we should log it in case of error, but I agree with @duglin, I thinks its safe to go on (and not fail right away).